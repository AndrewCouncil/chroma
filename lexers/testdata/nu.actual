"12" | into int

date now | date to-timezone "Europe/London"

{'name': 'nu', 'stars': 5, 'language': 'Python'} | upsert language 'Rust'

[one two three] | to yaml

[[framework, language]; [Django, Python] [Laravel, PHP]]

[{name: 'Robert' age: 34 position: 'Designer'}
 {name: 'Margaret' age: 30 position: 'Software Developer'}
 {name: 'Natalie' age: 50 position: 'Accountant'}
] | select name position

let name = "Alice"
$"greetings, ($name)!"
# => greetings, Alice!

let string_list = "one,two,three" | split row ","
$string_list
# => ╭───┬───────╮
# => │ 0 │ one   │
# => │ 1 │ two   │
# => │ 2 │ three │
# => ╰───┴───────╯

"Hello, world!" | str contains "o, w"
# => true

let str_list = [zero one two]
$str_list | str join ','
# => zero,one,two

'Hello World!' | str substring 4..8
# => o Wor

'Nushell 0.80' | parse '{shell} {version}'

"acronym,long\nAPL,A Programming Language" | from csv

$'(ansi purple_bold)This text is a bold purple!(ansi reset)'
# => This text is a bold purple!

[foo bar baz] | insert 1 'beeze'

[1, 2, 3, 4] | update 1 10

let numbers = [1, 2, 3]
$numbers | prepend 0

let numbers = [1, 2, 3]
$numbers | append 4

[cammomile marigold rose forget-me-not] | first 2

let planets = [Mercury Venus Earth Mars Jupiter Saturn Uranus Neptune]
$planets | each { |elt| $"($elt) is a planet of the solar system" }

$planets | enumerate | each { |elt| $"($elt.index + 1) - ($elt.item)" }

let scores = [3 8 4]
$"total = ($scores | reduce { |elt, acc| $acc + $elt })"
# => total = 15

def "main run" [] {
    print "running"
}

def "main build" [] {
    print "building"
}

def main [x: string] {
  $"Hello ($x | describe) ($x)"
}
