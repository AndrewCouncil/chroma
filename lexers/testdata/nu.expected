[
  {"type":"LiteralStringDouble","value":"\"12\""},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"into"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"int"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"date"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"now"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"date"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"to"},
  {"type":"Text","value":"-timezone"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringDouble","value":"\"Europe/London\""},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Operator","value":"{"},
  {"type":"LiteralStringSingle","value":"'name'"},
  {"type":"Text","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringSingle","value":"'nu'"},
  {"type":"Text","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringSingle","value":"'stars'"},
  {"type":"Text","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"5"},
  {"type":"Text","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringSingle","value":"'language'"},
  {"type":"Text","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringSingle","value":"'Python'"},
  {"type":"Operator","value":"}"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"upsert"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"language"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringSingle","value":"'Rust'"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Operator","value":"["},
  {"type":"Text","value":"one"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"two"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"three"},
  {"type":"Operator","value":"]"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"to"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"yaml"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Operator","value":"[["},
  {"type":"Text","value":"framework,"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"language"},
  {"type":"Operator","value":"]"},
  {"type":"Punctuation","value":";"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"["},
  {"type":"Text","value":"Django,"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"Python"},
  {"type":"Operator","value":"]"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"["},
  {"type":"Text","value":"Laravel,"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"PHP"},
  {"type":"Operator","value":"]]"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Operator","value":"[{"},
  {"type":"Text","value":"name:"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringSingle","value":"'Robert'"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"age:"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"34"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"position:"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringSingle","value":"'Designer'"},
  {"type":"Operator","value":"}"},
  {"type":"TextWhitespace","value":"\n "},
  {"type":"Operator","value":"{"},
  {"type":"Text","value":"name:"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringSingle","value":"'Margaret'"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"age:"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"30"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"position:"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringSingle","value":"'Software Developer'"},
  {"type":"Operator","value":"}"},
  {"type":"TextWhitespace","value":"\n "},
  {"type":"Operator","value":"{"},
  {"type":"Text","value":"name:"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringSingle","value":"'Natalie'"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"age:"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"50"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"position:"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringSingle","value":"'Accountant'"},
  {"type":"Operator","value":"}"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Operator","value":"]"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"select"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"name"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"position"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"let"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"name"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringDouble","value":"\"Alice\""},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"LiteralStringDouble","value":"$\"greetings, "},
  {"type":"LiteralStringInterpol","value":"("},
  {"type":"NameVariable","value":"$name"},
  {"type":"LiteralStringInterpol","value":")"},
  {"type":"LiteralStringDouble","value":"!\""},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"CommentSingle","value":"# =\u003e greetings, Alice!\n"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Keyword","value":"let"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"string_list"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringDouble","value":"\"one,two,three\""},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"split"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"row"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringDouble","value":"\",\""},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"NameVariable","value":"$string_list"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"CommentSingle","value":"# =\u003e ╭───┬───────╮\n# =\u003e │ 0 │ one   │\n# =\u003e │ 1 │ two   │\n# =\u003e │ 2 │ three │\n# =\u003e ╰───┴───────╯\n"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"LiteralStringDouble","value":"\"Hello, world!\""},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"str"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"contains"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringDouble","value":"\"o, w\""},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"CommentSingle","value":"# =\u003e true\n"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Keyword","value":"let"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"str_list"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"["},
  {"type":"Text","value":"zero"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"one"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"two"},
  {"type":"Operator","value":"]"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"NameVariable","value":"$str_list"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"str"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"join"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringSingle","value":"','"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"CommentSingle","value":"# =\u003e zero,one,two\n"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"LiteralStringSingle","value":"'Hello World!'"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"str"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"substring"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"4"},
  {"type":"Text","value":"..8"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"CommentSingle","value":"# =\u003e o Wor\n"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"LiteralStringSingle","value":"'Nushell 0.80'"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"parse"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringSingle","value":"'{shell} {version}'"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"LiteralStringDouble","value":"\"acronym,long\\nAPL,A Programming Language\""},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"from"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"csv"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"LiteralStringSingle","value":"$'(ansi purple_bold)This text is a bold purple!(ansi reset)'"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"CommentSingle","value":"# =\u003e This text is a bold purple!\n"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Operator","value":"["},
  {"type":"Text","value":"foo"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"bar"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"baz"},
  {"type":"Operator","value":"]"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"insert"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"1"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringSingle","value":"'beeze'"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Operator","value":"["},
  {"type":"LiteralNumber","value":"1"},
  {"type":"Text","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"2"},
  {"type":"Text","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"3"},
  {"type":"Text","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"4"},
  {"type":"Operator","value":"]"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"update"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"1"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"10"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"let"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"numbers"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"["},
  {"type":"LiteralNumber","value":"1"},
  {"type":"Text","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"2"},
  {"type":"Text","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"3"},
  {"type":"Operator","value":"]"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"NameVariable","value":"$numbers"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"prepend"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"0"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"let"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"numbers"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"["},
  {"type":"LiteralNumber","value":"1"},
  {"type":"Text","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"2"},
  {"type":"Text","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"3"},
  {"type":"Operator","value":"]"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"NameVariable","value":"$numbers"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"append"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"4"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Operator","value":"["},
  {"type":"Text","value":"cammomile"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"marigold"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"rose"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"forget-me-not"},
  {"type":"Operator","value":"]"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"first"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"2"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"let"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"planets"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"["},
  {"type":"Text","value":"Mercury"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"Venus"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"Earth"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"Mars"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"Jupiter"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"Saturn"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"Uranus"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"Neptune"},
  {"type":"Operator","value":"]"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"NameVariable","value":"$planets"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"each"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"{"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"Text","value":"elt"},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringDouble","value":"$\""},
  {"type":"LiteralStringInterpol","value":"("},
  {"type":"NameVariable","value":"$elt"},
  {"type":"LiteralStringInterpol","value":")"},
  {"type":"LiteralStringDouble","value":" is a planet of the solar system\""},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"}"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"NameVariable","value":"$planets"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"enumerate"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"each"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"{"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"Text","value":"elt"},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringDouble","value":"$\""},
  {"type":"LiteralStringInterpol","value":"("},
  {"type":"NameVariable","value":"$elt"},
  {"type":"Text","value":".index"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"+"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"1"},
  {"type":"LiteralStringInterpol","value":")"},
  {"type":"LiteralStringDouble","value":" - "},
  {"type":"LiteralStringInterpol","value":"("},
  {"type":"NameVariable","value":"$elt"},
  {"type":"Text","value":".item"},
  {"type":"LiteralStringInterpol","value":")"},
  {"type":"LiteralStringDouble","value":"\""},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"}"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"let"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"scores"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"["},
  {"type":"LiteralNumber","value":"3"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"8"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumber","value":"4"},
  {"type":"Operator","value":"]"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"LiteralStringDouble","value":"$\"total = "},
  {"type":"LiteralStringInterpol","value":"("},
  {"type":"NameVariable","value":"$scores"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"reduce"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"{"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"Text","value":"elt,"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"acc"},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"$acc"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"+"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameVariable","value":"$elt"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"}"},
  {"type":"LiteralStringInterpol","value":")"},
  {"type":"LiteralStringDouble","value":"\""},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"CommentSingle","value":"# =\u003e total = 15\n"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Keyword","value":"def"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringDouble","value":"\"main run\""},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"[]"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"{"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"Keyword","value":"print"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringDouble","value":"\"running\""},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Operator","value":"}"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"def"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringDouble","value":"\"main build\""},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"[]"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"{"},
  {"type":"TextWhitespace","value":"\n    "},
  {"type":"Keyword","value":"print"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringDouble","value":"\"building\""},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Operator","value":"}"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"def"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"main"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"["},
  {"type":"Text","value":"x:"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"string"},
  {"type":"Operator","value":"]"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"{"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"LiteralStringDouble","value":"$\"Hello "},
  {"type":"LiteralStringInterpol","value":"("},
  {"type":"NameVariable","value":"$x"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"describe"},
  {"type":"LiteralStringInterpol","value":")"},
  {"type":"LiteralStringDouble","value":" "},
  {"type":"LiteralStringInterpol","value":"("},
  {"type":"NameVariable","value":"$x"},
  {"type":"LiteralStringInterpol","value":")"},
  {"type":"LiteralStringDouble","value":"\""},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Operator","value":"}"},
  {"type":"TextWhitespace","value":"\n"}
]
